src/reader/error_impl.rs

```diff
@@ -0,0 +1,102 @@
+
+use std::io;
+use std::borrow::Cow;
+use std::fmt;
+use std::error;
+
+use util;
+use common::{Position, TextPosition};
+
+use super::{Error, ErrorKind};
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{} {}", self.pos, self.msg())
+    }
+}
+
+impl Position for Error {
+    #[inline]
+    fn position(&self) -> TextPosition { self.pos }
+}
+
+impl Error {
+    /// Returns a reference to a message which is contained inside this error.
+    #[inline]
+    pub fn msg(&self) -> &str {
+    	use super::ErrorKind::*;
+    	match self.kind {
+    		UnexpectedEof => &"Unexpected EOF",
+    		Utf8( ref reason ) => error_description( reason ),
```

Comment:
This is a minor stylistic issue, but spaces inside parentheses is not idiomatic style in Rust programs. This applies for the enum declaration as well.

I should definitely check how `rustfmt` works, probably it makes sense to use it for xml-rs.